arroyo-worker/src/connectors/polling_http.rs

```diff
@@ -0,0 +1,260 @@
+                .expect("could not construct http client"),
+            format: config
+                .format
+                .expect("polling http source must have a format configured"),
+            endpoint: url::Url::from_str(&table.endpoint).expect("invalid endpoint"),
+            method: match table.method {
+                None | Some(Method::Get) => reqwest::Method::GET,
+                Some(Method::Post) => reqwest::Method::POST,
+                Some(Method::Put) => reqwest::Method::PUT,
+                Some(Method::Patch) => reqwest::Method::PATCH,
+            },
+            body: table.body.map(|b| b.into()),
+            polling_interval: table
+                .poll_interval_ms
+                .map(|d| Duration::from_millis(d as u64))
+                .unwrap_or(DEFAULT_POLLING_INTERVAL),
+            emit_behavior: table.emit_behavior.unwrap_or(EmitBehavior::All),
+            _t: PhantomData,
+        }
+    }
+
+    fn name(&self) -> String {
+        "PollingHttpSource".to_string()
+    }
+
+    fn tables(&self) -> Vec<TableDescriptor> {
+        vec![arroyo_state::global_table("s", "polling http source state")]
+    }
+
+    async fn on_start(&mut self, ctx: &mut Context<(), T>) {
+        let s: GlobalKeyedState<(), PollingHttpSourceState<T>, _> =
+            ctx.state.get_global_keyed_state('s').await;
+
+        if let Some(state) = s.get(&()) {
+            self.state = state.clone();
+        }
+    }
+
+    async fn our_handle_control_message(
+        &mut self,
+        ctx: &mut Context<(), T>,
+        msg: Option<ControlMessage>,
+    ) -> Option<SourceFinishType> {
+        match msg? {
+            ControlMessage::Checkpoint(c) => {
+                debug!("starting checkpointing {}", ctx.task_info.task_index);
+                let state = self.state.clone();
+                let mut s: GlobalKeyedState<(), PollingHttpSourceState<T>, _> =
+                    ctx.state.get_global_keyed_state('s').await;
+                s.insert((), state).await;
+
+                if self.checkpoint(c, ctx).await {
+                    return Some(SourceFinishType::Immediate);
+                }
+            }
+            ControlMessage::Stop { mode } => {
+                info!("Stopping polling http source: {:?}", mode);
+
+                match mode {
+                    StopMode::Graceful => {
+                        return Some(SourceFinishType::Graceful);
+                    }
+                    StopMode::Immediate => {
+                        return Some(SourceFinishType::Immediate);
+                    }
+                }
+            }
+            ControlMessage::Commit { epoch: _ } => {
+                unreachable!("sources shouldn't receive commit messages");
+            }
+        }
+        None
+    }
+
+    async fn request(&mut self) -> Result<T, UserError> {
+        let mut request = self
+            .client
+            .request(self.method.clone(), self.endpoint.clone());
```

Comment:
I'm wondering if there should be a timeout and a maximum body size

Reply:
There's a 5 second timeout, which could be made configurable. A max response size is probably a good idea, but not straightforward to implement with reqwest.

Reply:
added a 5MB response size limit